extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -558,25 +558,34 @@ export class AmpA4A extends AMP.BaseElement {
     // So long as any one of these signing services can verify the
     // signature, then the creative is valid AMP.
     const keyInfoSetPromises = this.win.ampA4aValidationKeys;
+    // Track if verification found as it will ensure that promises yet to
+    // resolve will "cancel" as soon as possible saving unnecessary resource
+    // allocation.
+    let verificationFound = false;
+    const alreadyVerified = () => {
+      return verificationFound ?
+          Promise.reject('verification already found') : null;
+    };
     return some(keyInfoSetPromises.map(keyInfoSetPromise => {
       // Resolve Promise into Array of Promises of signing keys.
       return keyInfoSetPromise.then(keyInfoSet => {
```

Comment:
Array.prototype.map is not an async operation but instead executes the given function on each item in the array and returns an array whose value is the result of the given function in order of the items in the array.   Given its not async, I see no need to recheck already verified.